src/composables/useFetch.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ref,
+  watchEffect,
+  toValue,
+  ComputedRef,
+  type Ref,
+  type UnwrapRef,
+} from "vue";
+
+type UseFetchReturnType<T> = {
+  data: Ref<UnwrapRef<T> | null>;
+  error: Ref<UnwrapRef<Error> | null>;
+};
+
+export function useFetch<T>(
```

Comment:
nice that you created a composable

this is very reactish way of handling fetch, in vue I'd just created a composable with a method to which we can pass the url and expose it so we can call it from other component method or some lifecycle hook